Common/src/gRPC/Services/Submitter.cs

```diff
@@ -703,32 +709,12 @@ private IObjectStorage PayloadStorage(string session)
                                                         .ConfigureAwait(false));
     }
 
-    var taskDataModels = requests.Select(request =>
-                                         {
-                                           var intersect = parentExpectedOutputKeys.Intersect(request.ExpectedOutputKeys)
-                                                                                   .ToList();
-
-                                           var resultModel = request.ExpectedOutputKeys.Except(intersect)
-                                                                    .Select(key => new Result(session,
-                                                                                              key,
-                                                                                              request.Id,
-                                                                                              ResultStatus.Created,
-                                                                                              new List<string>(),
-                                                                                              DateTime.UtcNow,
-                                                                                              Array.Empty<byte>()));
-
-                                           return (Result: resultModel, Req: new IResultTable.ChangeResultOwnershipRequest(intersect,
-                                                                                                                           request.Id));
-                                         });
-
+    var taskDataModels = requests.Select(request => new IResultTable.ChangeResultOwnershipRequest(parentExpectedOutputKeys.Intersect(request.ExpectedOutputKeys),
```

Comment:
I know that it was already there, but this is accidentally quadratic because of the intersection of lists. It should most likely be an intersection of HashSet or something like that.

Reply:
We can change it in a perf oriented pr